src/features/codeActionProvider.ts

```diff
@@ -112,20 +114,62 @@ export default class CodeActionProvider extends AbstractProvider implements vsco
 
                 let edit = new vscode.WorkspaceEdit();
 
+                let fileToOpen: Uri = null;
+                let renamedFiles: Uri[] = [];
+
                 for (let change of response.Changes) {
-                    let uri = vscode.Uri.file(change.FileName);
-                    let edits: vscode.TextEdit[] = [];
-                    for (let textChange of change.Changes) {
-                        edits.push(vscode.TextEdit.replace(toRange2(textChange), textChange.NewText));
+                    if (change.ModificationType == FileModificationType.Renamed)
+                    {
+                        // The file was renamed. Omnisharp has already persisted
+                        // the right changes to disk. We don't need to try to
+                        // apply text changes.
+                        renamedFiles.push(Uri.file(change.FileName));
+                    }
+
+                    if (change.ModificationType == FileModificationType.Opened)
+                    {
+                        // The CodeAction requested that we open a file. 
+                        // Record that file name and keep processing CodeActions.
+                        // If a CodeAction requests that we open multiple files 
+                        // we only open the last one (what would it mean to open multiple files?)
+                        fileToOpen = vscode.Uri.file(change.FileName);
                     }
 
-                    edit.set(uri, edits);
+                    if (change.ModificationType == FileModificationType.Modified)
+                    {
+                        let uri = vscode.Uri.file(change.FileName);
```

Comment:
Based on your comment when `FileModificationType.Renamed` is true, should we check to see if this file is included in `renamedFiles`? And if so, does that change the order in which we loop through the changes?

Reply:
Right. We should probably collect all the renames first, so we don't try to edit files that got renamed.